src/messaging/ReliableMessageManager.h

```diff
@@ -0,0 +1,113 @@
+/*
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the classes corresponding to CHIP reliable message
+ *      protocol.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include <messaging/ReliableMessageProtocolConfig.h>
+
+#include <core/CHIPError.h>
+#include <system/SystemLayer.h>
+#include <system/SystemPacketBuffer.h>
+#include <system/SystemTimer.h>
+
+namespace chip {
+namespace messaging {
+
+class ReliableMessageContext;
+
+class ReliableMessageManager
+{
+private:
+    /**
+     *  @class RetransTableEntry
+     *
+     *  @brief
+     *    This class is part of the CHIP Reliable Messaging Protocol and is used
+     *    to keep track of CHIP messages that have been sent and are expecting an
+     *    acknowledgment back. If the acknowledgment is not received within a
+     *    specific timeout, the message would be retransmitted from this table.
+     *
+     */
+    class RetransTableEntry
```

Comment:
This should be a structu (all public data members). If a class, data members should be private with getters/setters, however that seems overkill.